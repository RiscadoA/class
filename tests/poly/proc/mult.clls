proc overengineered_forward<X>(x:~X, y:X){
    cut {
        println("send");
        send ch(x);
        println("lhs fwd");
        fwd ch y
        |ch: recv ~X; X|
        println("recv");
        recv ch(a);
        println("rhs fwd");
        fwd a ch
    }
};;

proc main() {
    cut {
        println("wait");
        wait x; ()
        |x:close|
        overengineered_forward<wait>(x, y)
        |y:close|
        println("close");
        close y
    }
};;
