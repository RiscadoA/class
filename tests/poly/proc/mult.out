close
send
lhs fwd
recv
rhs fwd
wait
